pkg/apis/operator/v1alpha1/types.go

```diff
@@ -526,6 +529,95 @@ type GardenerSchedulerConfig struct {
 	LogLevel *string `json:"logLevel,omitempty"`
 }
 
+// GardenerDashboardConfig contains configuration settings for the gardener-dashboard.
+type GardenerDashboardConfig struct {
+	// EnableTokenLogin specifies whether it is possible to log into the dashboard with a JWT token. If disabled, OIDC
+	// must be configured.
+	// +kubebuilder:default=true
+	// +optional
+	EnableTokenLogin *bool `json:"enableTokenLogin,omitempty"`
+	// FrontendConfigMapRef is the reference to a ConfigMap in the garden namespace containing the frontend
+	// configuration.
+	// +optional
+	FrontendConfigMapRef *corev1.LocalObjectReference `json:"frontendConfigMapRef,omitempty"`
+	// AssetsConfigMapRef is the reference to a ConfigMap in the garden namespace containing the assets (logos/icons).
+	// +optional
+	AssetsConfigMapRef *corev1.LocalObjectReference `json:"assetsConfigMapRef,omitempty"`
+	// GitHub contains configuration for the GitHub ticketing feature.
+	// +optional
+	GitHub *DashboardGitHub `json:"gitHub,omitempty"`
+	// LogLevel is the configured log level. Must be one of [trace,debug,info,warn,error].
+	// Defaults to info.
+	// +kubebuilder:validation:Enum=trace;debug;info;warn;error
+	// +kubebuilder:default=info
+	// +optional
+	LogLevel *string `json:"logLevel,omitempty"`
+	// OIDC contains configuration for the OIDC provider. This field must be provided when EnableTokenLogin is false.
+	// +optional
+	OIDC *DashboardOIDC `json:"oidcConfig,omitempty"`
+	// Terminal contains configuration for the terminal settings.
+	// +optional
+	Terminal *DashboardTerminal `json:"terminal,omitempty"`
+}
+
+// DashboardGitHub contains configuration for the GitHub ticketing feature.
+type DashboardGitHub struct {
+	// APIURL is the URL to the GitHub API.
+	// +kubebuilder:default=`https://api.github.com`
+	// +kubebuilder:validation:MinLength=1
+	APIURL string `json:"apiURL"`
+	// Organisation is the name of the GitHub organisation.
+	// +kubebuilder:validation:MinLength=1
+	Organisation string `json:"organisation"`
+	// Repository is the name of the GitHub repository.
+	// +kubebuilder:validation:MinLength=1
+	Repository string `json:"repository"`
+	// SecretRef is the reference to a secret in the garden namespace containing the GitHub credentials.
+	SecretRef corev1.LocalObjectReference `json:"secretRef"`
+	// PollInterval is the interval of how often the GitHub API is polled for issue updates. This field is only
+	// meaningful when there is no GitHub webhook configuration, i.e., when the referenced secret does not contain the
+	// 'webhookSecret' key. In this case, if this field is not provided, it will be implicitly defaulted with `15m`.
```

Comment:
```suggestion
	// PollInterval is the interval of how often the GitHub API is polled for issue updates. This field is used as a
	// fallback mechanism to ensure state synchronization, even when there is a GitHub webhook configuration. If a
	// webhook event is missed or not successfully delivered, the polling will help catch up on any missed updates.
	// If this field is not provided and there is no 'webhookSecret' key in the referenced secret, it will be
	// implicitly defaulted to 15m.
```